### ISSUE TITLE
5v esp32

### ISSUE THREAD

**Issue by folksman** [positron96/LocoNetControlStation#17]
I have a esp32 who need 5v input to run. As far as i see your current version power the lolin32 lite with 3v?
What can i change on the board to get 5v ?

**Comment by positron96** [comment]
Hi. Yes, I was stupid enough to use a lolin32 module that does not have 5V input pin and had to put extra 3.3V regulator on the board. If your module can be powered from 5V directly, there is no need for 3.3V regulator on the PCB (U7 and accompanying C1, C7 and C8). 
However, I don't think the module's pinout will allow to plug another module in place of Lolin32 as all esp32 modules have incompatible pinouts. Most likely you'll need to use wires to attach a different module. You'd need both 5V and 3.3V lines from the board to connect to the module, 5V regulator on board will provide power to esp32 module, and 3.3V regulator on esp32 will provide power to loconet and other parts of board.

**Comment by folksman** [comment]
i be bit confused. i want wire a m5stack atom esp32 (pls check the pinout). Can you tell me how i can wire it to the board then and which components i have to remove or bridge then? your help will be great
Am 15. Dez. 2021, 17:23 +0100 schrieb Paul Melnikov ***@***.***>:

**Comment by positron96** [comment]
Look, this project is envisioned as a starting point for other projects with different goals, and/or different hardware. You case seems to be exactly that. So what you want is definitely possible, but this is not a beginner-level task. You'd need to make changes to the PCB (or at the very least fully understand it to make wiring changes) and to the firmware. I am not willing to provide that level of support for the hardware I have not seen nor worked with, as I don't have time or resources. Also, I can't teach electronics or programming.

What I can do is provide some guidance, but for the case of pinout I can't give you more information than is already exhaustively provided in the [schematic](https://oshwlab.com/positron96/loconet-command-station). It shows what pins on esp32 are needed for the board to work, and what functions they serve. The names used (e.g. GPIO17) are common to all ESP32s. This information is also provided in the firmware at the top of `main.cpp`.

Regarding M5 atom, a quick googling shows that there is barely enough pins on m5atom for the board to fully work. A bare minimum for LocoNet+2 DCC channels is 8 pins (6 digital+2 analog). Not all the pins required by the firmware are wired out on the m5, so you'd need to modify the firmware yo use available pins. Also, M5s have their own extra hardware on board, you'll need to modify the firmware to make use of that (i.e. screen and buttons). While that will be fun, again, this is not a beginner-level tutorial-style project.

**Comment by folksman** [comment]
Ok. As far as i see i can catch 5v on the U4 already. If i understand you right then i can´t remove U7 and accompanying C1, C7 and C8 since they are required by Loconet etc? 
For what exactly is Sense - Analog IN - 36 in only and In - 39 required?

**Comment by positron96** [comment]
Yes, you can take 5V from U4. Note that if you remove 3.3V regulator, you'll need to provide 3.3V from esp32 module to the board to power loconet comparator and hex inverter.

Pins 36 and 39 read current on DCC tracks and provide functions of reading CVs and overcurrent detection.

**Comment by positron96** [comment]
I'm converting this to discussion since it's not an issue topic.
